waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -855,7 +855,8 @@ method isReady*(g: OnchainGroupManager): Future[bool] {.async.} =
   g.retryWrapper(currentBlock, "Failed to get the current block number"):
     cast[BlockNumber](await g.ethRpc.get().provider.eth_blockNumber())
 
-  if g.latestProcessedBlock < currentBlock:
+  # the node is still able to process messages if it is behind the latest block by a factor of the valid roots
+  if uint(g.latestProcessedBlock) < uint(currentBlock) - uint(g.validRoots.len):
```

Comment:
I wonder if is safer to add parenthesis:
```suggestion
  if uint(g.latestProcessedBlock) < ( uint(currentBlock) - uint(g.validRoots.len) ):
```

Reply:
doesn't harm, i would add it as well :)